subsys/testsuite/busy_sim/busy_sim.c

```diff
@@ -169,7 +180,9 @@ static int busy_sim_init(const struct device *dev)
 	struct busy_sim_data *data = get_dev_data(dev);
 
 	if ((config->pin_spec.port && !device_is_ready(config->pin_spec.port)) ||
-	    !device_is_ready(config->counter) || !device_is_ready(config->entropy)) {
+	    !device_is_ready(config->counter) ||
+	    (!IS_ENABLED(CONFIG_XOSHIRO_RANDOM_GENERATOR) &&
+	    !device_is_ready(config->entropy))) {
```

Comment:
Since `config->entropy` is not used when `CONFIG_XOSHIRO_RANDOM_GENERATOR` is enabled, the field does not need to be even initialized then (currently, the initialization requires that the `zephyr,entropy` node is always present).